src/OrchardCore/OrchardCore.Abstractions/Shell/Models/FeatureProfile.cs

```diff
@@ -1,11 +1,20 @@
 using System.Collections.Generic;
 using System.ComponentModel.DataAnnotations;
+using Newtonsoft.Json.Linq;
 
 namespace OrchardCore.Environment.Shell.Models
 {
     public class FeatureProfile
     {
+        public string Id { get; set; }
+
+        public string Name { get; set; }
+
         public List<FeatureRule> FeatureRules = new List<FeatureRule>();
+
+        // should we inherit from Entity instead?
+        public JObject Properties { get; set; } = new JObject();
```

Comment:
Where is it used?

Reply:
Currently no where. I just thought it should exists to allow consumer to attach any custom parts to it should any be needed just like we do with Entity.